editor/plugins/theme_editor_plugin.cpp

```diff
@@ -1588,15 +1588,12 @@ void ThemeItemEditorDialog::_add_theme_item(Theme::DataType p_data_type, String
 
 void ThemeItemEditorDialog::_remove_theme_type(const String &p_theme_type) {
 	Ref<Theme> old_snapshot = edited_theme->duplicate();
-	Ref<Theme> new_snapshot = edited_theme->duplicate();
 
 	EditorUndoRedoManager *ur = EditorUndoRedoManager::get_singleton();
 	ur->create_action(TTR("Remove Theme Type"));
 
-	new_snapshot->remove_type(p_theme_type);
-
-	ur->add_do_method(*edited_theme, "clear");
-	ur->add_do_method(*edited_theme, "merge_with", new_snapshot);
+	// Avoid merging with new_snapshot, if type was empty new_snapshot will merge as empty data and cause all empty types to be deleted, instead of clicked type.
```

Comment:
The comment does not make sense with the new code. `new_snapshot` no longer exists.